src/components/CheckBoxList/check-box-list.tsx

```diff
@@ -0,0 +1,218 @@
+                                                style={{
+                                                    opacity:
+                                                        hover ===
+                                                            getValueId(item) &&
+                                                        !isDragDisable
+                                                            ? '1'
+                                                            : '0',
+                                                }}
+                                            >
+                                                <DragIndicatorIcon />
+                                            </IconButton>
+                                            <CheckBoxItem
+                                                item={item}
+                                                checked={isChecked(item)}
+                                                label={
+                                                    getValueLabel
+                                                        ? getValueLabel(item)
+                                                        : ''
+                                                }
+                                                onClick={() =>
+                                                    toggleSelection(
+                                                        getValueId(item)
+                                                    )
+                                                }
+                                                secondaryAction={handleSecondaryAction(
+                                                    item
+                                                )}
+                                                labelSx={labelSx}
+                                                disabled={
+                                                    isDisabled
+                                                        ? isDisabled(item)
+                                                        : false
+                                                }
+                                            />
+                                        </ListItem>
+                                    </Box>
+                                )}
+                            </Draggable>
+                        )}
+                        {!isCheckBoxDraggable && (
+                            <CheckBoxItem
+                                item={item}
+                                checked={isChecked(item)}
+                                label={getValueLabel(item)}
+                                onClick={() =>
+                                    toggleSelection(getValueId(item))
+                                }
+                                secondaryAction={handleSecondaryAction(item)}
+                                labelSx={labelSx}
+                                disabled={isDisabled(item)}
+                            />
```

Comment:
this block of code is duplicated if I'm not wrong, and I can see some small behaviour differences ( "label={getValueLabel ? getValueLabel(item) : '' }" becomes "label={getValueLabel(item)}")

create a high order component around <CheckBoxItem> that will wrap it with the Draggable behaviour or not depending on a prop ?